core/nixpkgs.bzl

```diff
@@ -162,22 +179,56 @@ def _nixpkgs_package_impl(repository_ctx):
     if repository_ctx.attr.nix_file and repository_ctx.attr.nix_file_content:
         fail("Specify one of 'nix_file' or 'nix_file_content', but not both.")
     elif repository_ctx.attr.nix_file:
-        nix_file = cp(repository_ctx, repository_ctx.attr.nix_file)
-        expr_args = [repository_ctx.path(nix_file)]
+        src = repository_ctx.attr.nix_file
+        cp(repository_ctx, src, label_to_path(src, prefix="nixpkg/src"))
```

Comment:
Could you explain what this change does and why it's needed? IIUC this introduces an extra prefix and nests the copied files deeper. Wouldn't that break cross workspace location expansion references?

Reply:
It's been a while, but I *think* the idea was that we want to have a `nixpkgs/default.nix` on every package, but then we need to prevent this interfering with user provided files, so by putting the user input under `nixpkgs/src`, we ensure there will be no conflict

Reply:
Hairiest merge res is around here. Didn't dig deep enough to figure out how to resolve- just took changes and hoped it worked

Reply:
I see, rules_nixpkgs [already creates a special sub-directory called `bazel-support`](https://github.com/tweag/rules_nixpkgs/blob/62d63740061dd94ecee3ec2c0cfc8a7ed409b98c/core/nixpkgs.bzl#L324), which houses the Nix output link used as GC-root.

So, user-code already has to avoid collision with that reserved name.

Would it be possible to generate the `nixpkgs/default.nix` under that sub-directory instead? Then this change wouldn't need to introduce the `nixpkgs/src` prefix.